### DISCUSSION TITLE
how to setup nvim-dap to work with multiple debuggers

### DISCUSSION THREAD

**Discussion by ipatch** [mfussenegger/nvim-dap#335] [Q&A]
hello 👋, i recently discovered this project after watching [this youtube video][lnk1]. i've looked through the examples in the wiki about setting up various debuggers for various languages. however, i didn't see an example or any mention of setting up nvim-dap to work with multiple debuggers. i've setup my neovim configuration to debug nodejs by attaching to the debugger when node is run with the `--inspect` flag, but i'd also like to be able to debug python scripts using the same manner of attaching to a debugger.

is there a crude example of how i could use the same keybinding to attach to the debugger depending on the current open buffer or possibly use and attach to the debugger based on filetype, or something of that nature. using my current setup if i use my key mapping to attach to the debugger after setting a breakpoint my configuration always attempts to attach to a node process, and not the python debugger _debugpy_.

i've taken a lot inspiration from reading through [these dotfiles][lnk2] but it appears he's only using dap to debug nodejs.

~~i'll link to my current setup here in a sec.~~

my current setup working with nvim-dap [here][lnk3]

[lnk1]: <https://www.youtube.com/watch?v=ga3Cas7vNCk>
[lnk2]: <https://github.com/David-Kunz/vim/blob/master/init.lua>
[lnk3]: <https://github.com/ipatch/dotfiles/blob/dev.summer/jobs/macos/Users/mr-fancy/config/nvim/init.lua#L150-L219>

**Comment by mfussenegger** [comment]
You should be able to launch a debug session in a python file with your `map('n', '<leader>dn', ':lua require"dap".continue()<CR>')` mapping.


You can also remove the `attach` from the `debugHelpers` and instead add it to your javascript configuration:


```lua
dap.configurations.javascript = {
  {
    types = 'node2',
    requrest = 'launch',
    program = '${workspaceFolder}/${file}',
    cwd = vim.fn.getcwd(),
    sourceMaps = true,
    protocol = 'inspector',
    console = 'integratedTerminal',
  },
  {
    type = 'node2',
    request = 'attach',
    cwd = vim.fn.getcwd(),
    sourceMaps = true,
    protocol = 'inspector',
    skipFiles = {'<node_internals>/**/*.js'},
  },
}
```

If you then use your `<leader>dn` mapping it will prompt you for which configuration to use.

**Reply by ipatch** [reply]
i'll definitely try this out here in a bit. thank you so much for posting an example. 🙏
